packages/p2p/src/components/my-ads/my-ads-table.jsx

```diff
@@ -60,31 +61,45 @@ const MyAdsTable = () => {
                         is_warn
                     />
                 )}
-                {floating_rate_store.change_ad_alert && (
-                    <div className='p2p-my-ads__warning'>
-                        <HintBox
-                            icon='IcAlertWarning'
-                            message={
-                                <Text as='p' size='xxxs' color='prominent' line_height='xs'>
-                                    {floating_rate_store.reached_target_date ? (
-                                        <Localize i18n_default_text='Your ads with fixed rates have been deactivated. Set floating rates to reactivate them.' />
-                                    ) : (
-                                        <Localize
-                                            i18n_default_text={
-                                                'Floating rates are enabled for {{local_currency}}. Ads with fixed rates will be deactivated. Switch to floating rates by {{end_date}}'
-                                            }
-                                            values={{
-                                                local_currency: local_currency || '',
-                                                end_date: floating_rate_store.fixed_rate_adverts_end_date || '',
-                                            }}
-                                        />
-                                    )}
-                                </Text>
-                            }
-                            is_warn
-                        />
-                    </div>
-                )}
+                {floating_rate_store.change_ad_alert &&
```

Comment:
Could this code be possibly refactored into something like this instead, so that to avoid less code lines, the function `generateHintBoxMessage` is placed outside the parent component to avoid mounting-remounting when returning a component:
https://levelup.gitconnected.com/code-review-avoid-declaring-react-component-inside-parent-component-1768a645f523
```
// put this function outside the parent component to avoid mounting-remounting
const generateHintBoxMessage = () => {
  if (floating_rate_store.rate_type === ad_type.FLOAT) {
    if (floating_rate_store.reached_target_date) {
      return (
        <Localize i18n_default_text="Your ads with fixed rates have been deactivated. Set floating rates to reactivate them." />
      );
    }
    return (
      <Localize
        i18n_default_text={
          "Floating rates are enabled for {{local_currency}}. Ads with fixed rates will be deactivated. Switch to floating rates by {{end_date}}"
        }
        values={{
          local_currency:
            general_store.client.local_currency_config.currency || "-",
          end_date: floating_rate_store.fixed_rate_adverts_end_date || "-",
        }}
      />
    );
  }
  return (
    <Localize i18n_default_text="Your ads with floating rates have been deactivated. Set fixed rates to reactivate them." />
  );
};

{
  floating_rate_store.change_ad_alert && (
    <div className="p2p-my-ads__warning">
      <HintBox
        icon="IcAlertWarning"
        message={
          <Text as="p" size="xxxs" color="prominent" line_height="xs">
            {generateHintBoxMessage()}
          </Text>
        }
        is_warn
      />
    </div>
  );
}
```

Reply:
The components such as Localize or Hintbox are imported into MyAdsTable component. So there wont be the issue of mounting and un-mounting.  Creating this outside might help with modularity but the function will not be reused. Hence I think it is unnecessary. @carol-binary, what are your thoughts on this?

Reply:
Will refactor it.